feat(docs): migration documentations

jw-foss opened the pull request:
- Move documentations to docs/

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

jw-foss committed:
feat(docs): migration documentations

- Move documentations to docs/

JeremyWuuuuu commented:
Take https://github.com/element-plus/element-plus/pull/3283/files#diff-96b935734c9c79d200aba1a168b1b734d61d2ab86cb13d8aa1b51c61ec71a3a8 as an example here:

![image](https://user-images.githubusercontent.com/15975785/132443396-517e54a6-6070-4419-8ace-098b9151649c.png)

1. The file path needs to be co-responsible to example files.
2. When encountering demo without `template` block, `template` is required to be added to the demo.

@0song Please take file from `affix.md` to `container.md`
@adaex Please take file from `date-picker.md` to `i18n.md`
@imswk Please take file from `image.md` to `pagination.md`
@msidolphin Please take file from `popconfirm.md` to `slider.md`

jw-foss committed:
remove unused files

msidolphin committed:
docs: [popconfirm] migrate document (#3285)

jw-foss committed:
feat(docs): migrate documentation from space.md to upload.md (#3292)

- Upload documentations acorrodingly

adaex committed:
docs: [date-picker] migrate document  (#3289)

* docs: [date-picker] migrate document

* fix: typo

msidolphin committed:
docs: [slider] migrate document (#3287)

adaex committed:
docs: migrate documents (#3290)

* docs: migrate-datetime-picker

* docs: migrate descriptions

* docs: migrate dialog

* docs: migrate divider

* docs: migrate drawer

* docs: migrate drapdown

* docs: fix drapdown

* docs: migrate empty

* docs: migrate form

* docs: add scoped for style

* docs: simplify toRefs

0song committed:
chore: update doc (#3297)

* chore: update doc

* chore: update doc

* chore: update doc

jw-foss committed:
feat(docs): migrate documentations from infinite-scroll to page-header (#3303)

- Update docs accordingly
- Update CodePen icon to match style
- Update component name to match others

msidolphin committed:
docs: migrate documentions (#3293)

migrate list:
* image
* popover
* scrollbar
* radio
* rate
* skeleton
* select
* select-v2
* reault
* progress
* pagination

jw-foss committed:
chore: update doc (#3305)

Co-authored-by: 0song <0song@gmail.com>

jw-foss committed:
Fix broken pipeline

zouhangwithsweet committed:
chore: update demo plugin

jw-foss committed:
website perfection

jw-foss committed:
fix hydration bug

sxzz committed:
docs: update guide (#3342)

* WIP docs

* docs: update docs

jw-foss committed:
add dark mode

jw-foss committed:
make dev fetch components from local instead of node_modules
